### PR TITLE
fix: escape vite asset import suffixes

### PR DESCRIPTION
**Pull request by barakcodes**
Add the regex to exclude all vite asset import suffixes, might be better done in the @hono/vite-dev-server(maybe), but this fixes the issue for now

Rel #44

### CODE REVIEW

**Comment by barakcodes** [comment]
> Thanks, I checked this out locally and I still see the 404 in the example you updated.

Hmm, it works for me. I look more into it

**Comment by barakcodes** [comment]
> Thanks, I checked this out locally and I still see the 404 in the example you updated.

@john-griffin could you give it another try?

**Comment by barakcodes** [comment]
> Still a 404 with that pattern. I'm running via the `npm run dev` command if that makes any difference, not `npm run start-with-adapter`

shoot! I'll take a deeper look at this over the weekend

File: src/vite-plugin.ts [hunk 1]
```diff
@@ -31,6 +31,8 @@ export default (options: Options): Plugin => {
       '/assets/**',
       '/app/**',
       '/src/app/**',
+      // matches for vite's import assets suffixes
+      /\?(inline|url|no-inline|raw)$/
```
[the remark this answers is not in the document]
Comment (yusukebe) [reply src/vite-plugin.ts:35]: You can refer to this error message:

CleanShot 2024-12-05 at 19 36 57@2x

**Comment by barakcodes** [comment]
So 

> Still a 404 with that pattern. I'm running via the `npm run dev` command if that makes any difference, not `npm run start-with-adapter`

turns out I was testing with `?inline` which did not have`?import` prefix . this should now work 
as it matches  both `?inline` and `?import&inline` and similar combinations

**Comment by yusukebe** [comment]
Hey @barakcodes 

Thank you for fixing. I've added the test for this change.

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
Looks good! Let's land it.

**Pull request merged**